File: scratch/update_verified.py
import csv

file_path = 'data/postgres/v2-data/listings.csv'

rows = []
header = []
updated = 0
with open(file_path, mode='r', newline='', encoding='utf-8') as f:
    reader = csv.reader(f)
    header = next(reader)
    
    verified_idx = header.index('verified')
    
    for row in reader:
        if row[verified_idx] in ['true', 't', '1', 'TRUE']:
            row[verified_idx] = 'false'
            updated += 1
        elif row[verified_idx] == '':
            row[verified_idx] = 'false'
            updated += 1
        rows.append(row)

with open(file_path, mode='w', newline='', encoding='utf-8') as f:
    writer = csv.writer(f)
    writer.writerow(header)
    writer.writerows(rows)

print(f"Updated {updated} records to verified = false in listings.csv")

File: scratch/update_verified.go
package main

import (
	"encoding/csv"
	"fmt"
	"os"
)

func main() {
	filePath := "data/postgres/v2-data/listings.csv"
	file, err := os.Open(filePath)
	if err != nil {
		panic(err)
	}
	defer file.Close()

	reader := csv.NewReader(file)
	records, err := reader.ReadAll()
	if err != nil {
		panic(err)
	}

	if len(records) == 0 {
		fmt.Println("No records found")
		return
	}

	header := records[0]
	verifiedIdx := -1
	for i, h := range header {
		if h == "verified" {
			verifiedIdx = i
			break
		}
	}

	if verifiedIdx == -1 {
		fmt.Println("verified column not found")
		return
	}

	updated := 0
	for i := 1; i < len(records); i++ {
		val := records[i][verifiedIdx]
		if val == "true" || val == "t" || val == "1" || val == "TRUE" || val == "" {
			records[i][verifiedIdx] = "false"
			updated++
		}
	}

	outFile, err := os.Create(filePath)
	if err != nil {
		panic(err)
	}
	defer outFile.Close()

	writer := csv.NewWriter(outFile)
	err = writer.WriteAll(records)
	if err != nil {
		panic(err)
	}
	writer.Flush()

	fmt.Printf("Updated %d records to verified = false in listings.csv\n", updated)
}
